The build system includes its one check for available CMake build types.
This check is incorrect as it doesn't include the None build type.

See: https://github.com/Z3Prover/z3/issues/3071

diff -upr z3-z3-4.8.7.orig/CMakeLists.txt z3-z3-4.8.7/CMakeLists.txt
--- z3-z3-4.8.7.orig/CMakeLists.txt	2020-02-21 23:54:37.917997553 +0100
+++ z3-z3-4.8.7/CMakeLists.txt	2020-02-21 23:55:22.061925859 +0100
@@ -129,7 +129,7 @@ set(Z3_DEPENDENT_EXTRA_CXX_LINK_FLAGS ""
 # Build type
 ################################################################################
 message(STATUS "CMake generator: ${CMAKE_GENERATOR}")
-set(available_build_types Debug Release RelWithDebInfo MinSizeRel)
+set(available_build_types None Debug Release RelWithDebInfo MinSizeRel)
 if (DEFINED CMAKE_CONFIGURATION_TYPES)
   # Multi-configuration build (e.g. Visual Studio and Xcode). Here
   # CMAKE_BUILD_TYPE doesn't matter
